Strategy/IFileReader.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WeatherMonitoringSystem.model;
+
+namespace WeatherMonitoringSystem.Strategy
+{
+    public interface IFileReader
+    {
+        WeatherData Read(string data);
```

Comment:
We can use this interface just as a wrapper to unify other serializers.

```suggestion
        T? Deserialize<T>(string data);
```

That way we can create `JsonSerializer` and `XmlSerializer` that are wrappers to `System.Text.Json.JsonSerializer` and `System.Xml.Serialization.XmlSerializer`. But those are wrappers that inherit our `ISerializer` interface, and thus we can use multiple serializers under the same interface/type.